### ISSUE TITLE
netcdf files need a title attribute to be read?

### ISSUE THREAD

**Issue by CharlieLaughton** [Amber-MD/cpptraj#123]
I'm trying to read netcdf trajectory files created by MDAnalysis into cpptraj, but run into problems:
```
CPPTRAJ: Trajectory Analysis. V14.22
    ___  ___  ___  ___
     | \/ | \/ | \/ | 
    _|_/\_|_/\_|_/\_|_
        Reading 'penta.top' as Amber Topology
> trajin mdanalout.ncdf
        Reading 'mdanalout.ncdf' as Amber NetCDF
NETCDF Error: NetCDF: Attribute not found
Error: Getting length for attribute title
NETCDF Error: NetCDF: Variable not found
Error: Getting spatial VID
Error: Could not set up mdanalout.ncdf for reading.
Error: Could not set up input trajectory 'mdanalout.ncdf'.
```

This seems to suggest that the missing "title" attribute in MDAnalysis-generated .ncdf files is an issue. But as far as I can see from the "official" Amber netcdf format definition, the "title" attribute is not compulsory - is this right?
(I should mention that the generated .ncdf file seems otherwise OK - it reads back into MDAnalysis no problem)

**Comment by swails** [comment]
> This seems to suggest that the missing "title" attribute in MDAnalysis-generated .ncdf files is an issue.

This is actually misleading.  While it says "error", this is actually non-fatal in the code as far as I can tell.  The *real* error comes later:

```
NETCDF Error: NetCDF: Variable not found
Error: Getting spatial VID
```

In other words, the "spatial" variable is missing (this variable should always be set to 'x', 'y', 'z').  According to the NetCDF specification, this variable is *not* optional, and so MDAnalysis does not technically write valid NetCDF trajectory files.

The only thing that MDAnalysis being able to read the trajectory file means is that they ignore this variable (assuming it to be 'x', 'y', 'z'), but their writer is still non-conforming.

**Comment by drroe** [comment]
> While it says "error", this is actually non-fatal in the code as far as I can tell. 

Yes, this should be changed to a warning.

> MDAnalysis does not technically write valid NetCDF trajectory files.

I'm of two minds about what to do here. One the one hand forgetting to define the 'spatial' variable could be considered minor, and since presumably the "important" data (coordinates, velocities, box info) is present, and you can just assume things are XYZ and continue with a warning. On the other hand, we want to encourage adherence to the specification.

**Comment by CharlieLaughton** [comment]
Thanks for this. I think adhering to a spec is fair enough - since it seems the ball is probably in the court of the MDAnalysis team at this point, I'll raise this as an issue with them and point them at this discussion - OK?

**Comment by swails** [comment]
> Thanks for this. I think adhering to a spec is fair enough - since it seems the ball is probably in the court of the MDAnalysis team at this point, I'll raise this as an issue with them and point them at this discussion - OK?

Sounds good.  It sounds like @drroe may be in favor of modifying cpptraj to actually read this NetCDF file, but there are multiple readers out there that could flop when given this NetCDF file (possible candidates I can think of: ParmEd's reader, MDTraj's reader, VMD's reader, other readers in Amber like sander, and Chimera's reader, just to name a few).  So it's still probably good to raise an issue with MDAnalysis.

**Comment by drroe** [comment]
I've potentially fixed this with PR #127. Can you try it out and let me know if it works for you? Thanks.

**Comment by CharlieLaughton** [comment]
Yes this seems to work:
```
CPPTRAJ: Trajectory Analysis. V16.00b
    ___  ___  ___  ___
     | \/ | \/ | \/ | 
    _|_/\_|_/\_|_/\_|_

| Date/time: 10/14/15 21:50:05
| Available memory: 210.051 MB

        Reading 'spcbox.top' as Amber Topology
        Loading previous history from log 'cpptraj.log'
> trajin mdaspcbox.ncdf
        Reading 'mdaspcbox.ncdf' as Amber NetCDF
NetCDF: Attribute not found
Warning: Getting length for attribute 'title'
NetCDF: Variable not found
Warning: Could not get spatial VID. File may not be Amber NetCDF compliant.
Warning: Assuming spatial variables are 'x', 'y', 'z'
> 
```
(after this, actions on this file appear to be processed fine)

Thanks guys, I believe the MDAnalysis team are looking into this too.

**Comment by drroe** [comment]
@CharlieLaughton Great, thanks for testing!

Fixed by #127

**Comment by orbeckst** [comment]
That's our fault, MDAnalysis/mdanalysis#488.

Thanks for looking at it!
